### PR TITLE
[macro] fix printing of TExtend

### LINKED ISSUE #9405: Macro printer outputs invalid syntax for TExtend type
Example:

```hx
import haxe.macro.Printer;

class Main {
  public static function main() {
    final type = macro : {> Foo, bar: Int, baz: Int};
    final printer = new Printer();
    final print = printer.printComplexType(type);
    trace(print); // {> Foo, var bar : Int, var baz : Int }
  }
}
```

`{> Foo, var bar : Int, var baz : Int }` is not valid since it uses class notation for fields but separates them with comma instead of semicolon and lacks trailing semicolon.

### PR DESCRIPTION
**Pull request by vonagam**
Fixes #9405.

### CODE REVIEW

File: std/haxe/macro/Printer.hx [hunk 1]
```diff
@@ -138,7 +138,7 @@ class Printer {
 			case TParent(ct): "(" + printComplexType(ct) + ")";
 			case TOptional(ct): "?" + printComplexType(ct);
 			case TNamed(n, ct): n + ":" + printComplexType(ct);
-			case TExtend(tpl, fields): '{> ${tpl.map(printTypePath).join(" >, ")}, ${fields.map(printField).join(", ")} }';
+			case TExtend(tpl, fields): '{> ${tpl.map(printTypePath).join(" >, ")}, ${[for (f in fields) printField(f) + "; "].join("")}}';
```
Comment (Aurel300) [inline std/haxe/macro/Printer.hx]: The field fix is fine, but what's up with the `.join(" >, ")`? Surely that should be `.join(", > ")`…

```suggestion
			case TExtend(tpl, fields): '{> ${tpl.map(printTypePath).join(", > ")}, ${[for (f in fields) printField(f) + "; "].join("")}}';
```

File: tests/unit/src/unit/issues/Issue9405.hx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package unit.issues;
+
+class Issue9405 extends Test {
+  function test() {
+    var ct = macro : {> Foo, bar: Int, baz: Int};
+    var printer = new haxe.macro.Printer();
+    var s = printer.printComplexType(ct);
+    eq("{> Foo, var bar : Int; var baz : Int; }", s);
```
Comment (Aurel300) [inline tests/unit/src/unit/issues/Issue9405.hx]:
```suggestion
    var ct = macro : {> Foo, > Bar, bar: Int, baz: Int};
    var printer = new haxe.macro.Printer();
    var s = printer.printComplexType(ct);
    eq("{> Foo, > Bar, var bar : Int; var baz : Int; }", s);
```

Comment (Aurel300) [reply tests/unit/src/unit/issues/Issue9405.hx]: Could be a separate test I suppose…

File: std/haxe/macro/Printer.hx (hunk 1 above)
Comment (Gama11) [reply std/haxe/macro/Printer.hx]: This syntax requires a trailing `,` IIRC.

**Comment by vonagam** [comment]
Fixed the additional issue pointed out by @Aurel300.

Also now handles case when `TExtend` has an empty array of extended types (so same as usual `TAnonymous`).

File: std/haxe/macro/Printer.hx (hunk 1 above)
Comment (vonagam) [reply std/haxe/macro/Printer.hx]: Good catch @Aurel300.
@Gama11 trailing comma ensured after `join`, before fields.

**Pull request merged**